# 1. Đọc bộ dữ liệu, cho biết số dòng, số cột và kiểu dữ liệu của các thuộc tính.

In [1]:
import pandas as pd
import numpy as np
df=pd.read_csv('GDPlist.csv', header=0, encoding='ISO-8859-1')
display (df)

Country      Continent  GDP (millions of US$)
0     Afghanistan           Asia                  18181
1         Albania         Europe                  12847
2         Algeria         Africa                 190709
3          Angola         Africa                 100948
4       Argentina  South America                 447644
..            ...            ...                    ...
120    Uzbekistan           Asia                  45353
121     Venezuela  South America                 315841
122       Vietnam           Asia                 122722
123         Yemen         Africa                  33675
124        Zambia         Africa                  19206

[125 rows x 3 columns]

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 125 entries, 0 to 124
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Country                125 non-null    object
 1   Continent              125 non-null    object
 2   GDP (millions of US$)  125 non-null    int64 
dtypes: int64(1), object(2)
memory usage: 3.1+ KB


In [4]:
df.shape

(125, 3)

In [5]:
df.shape[0]

125

In [6]:
df.shape[1]

3

# 2. Tính giá trị lớn nhất và nhỏ nhất của GDP.

In [7]:
df_gdp=df.loc[:, 'GDP (millions of US$)']

In [8]:
df_gdp.min()

10600

In [9]:
df_gdp.max()

15094025

# 3. Hãy cho biết xu hướng phân bố dữ liệu của GDP.

In [11]:
df_gdp.median()

96089.0

In [12]:
df_gdp.mean()

555427.784

In [19]:
Q1_area=df_gdp.quantile(0.25, 'linear')
Q1_area 

25742.0

In [26]:
Q3_area=df_gdp.quantile(0.75)
Q3_area

333238.0

#### Trung bình lớn hơn trung vị, dữ liệu có xu hướng lệch phải
#### 75% các quốc gia có GDP<333238 USD
#### 25% các quốc gia có GDP>333238 USD
#### 25% các quốc qia có GDP<25742 USD

 # 4. Hãy cho biết châu lục nào xuất hiện nhiều nhất?

In [27]:
df.loc[:, 'Continent'].mode()

0    Europe
Name: Continent, dtype: object

# 5. Với mỗi châu lục hãy tính tổng GDP; trung bình cộng GDP. Hợp nhất 2 bảng này thành một bảng duy nhất gồm 3 thông tin: Tên châu lục; Tổng GDP; TBC GDP.

In [46]:
df_mean=df.pivot_table(values='GDP (millions of US$)', index='Continent', aggfunc='mean')
df_mean

GDP (millions of US$)
Continent                           
Africa                  6.278659e+04
Asia                    6.784970e+05
Europe                  5.172805e+05
North America           1.658041e+06
Oceania                 5.542423e+05
South America           4.601352e+05

In [47]:
df_sum=df.pivot_table(values='GDP (millions of US$)', index='Continent', aggfunc='sum')
df_sum

GDP (millions of US$)
Continent                           
Africa                       1820811
Asia                        24425891
Europe                      19139380
North America               18238447
Oceania                      1662727
South America                4141217

In [48]:
df1=pd.merge(df_mean,df_sum, on='Continent')
df1

GDP (millions of US$)_x  GDP (millions of US$)_y
Continent                                                      
Africa                    6.278659e+04                  1820811
Asia                      6.784970e+05                 24425891
Europe                    5.172805e+05                 19139380
North America             1.658041e+06                 18238447
Oceania                   5.542423e+05                  1662727
South America             4.601352e+05                  4141217